front/js/product.js

```diff
@@ -13,21 +12,75 @@ fetch("http://localhost:3000/api/products/"+ urlId)
     //get title of page
     document.getElementsByTagName("title").innerText = `${product.name}`;
     //get class item__img
-    document.querySelector(".item__img").innerHTML=`<img src="${product.imageUrl}" alt="${product.altTxt}">`;
+    document.querySelector(
+      ".item__img"
+    ).innerHTML = `<img src="${product.imageUrl}" alt="${product.altTxt}">`;
     //get id titile
     document.getElementById("title").innerText = `${product.name}`;
     //get id price
-    document.getElementById("price").innerText =`${product.price}`;
+    document.getElementById("price").innerText = `${product.price}`;
     //get id description
     document.getElementById("description").innerText = `${product.description}`;
-    //get color 
+    //get color
     product.colors.forEach((color) => {
-        document.getElementById("colors").innerHTML +=`<option value="${color}">${color}</option>`;
-      ;
+      document.getElementById(
+        "colors"
+      ).innerHTML += `<option value="${color}">${color}</option>`;
     });
+    addCart(product);
   })
   .catch((err) => {
     // Une erreur est survenue
-    const item = document.getElementById("item");
+    const item = document.getElementsByClassName("item")[0];
     item.innerText = `Une erreur est survenu (${err})`;
   });
+
+//................Cart button......................
+
+const addCart = (product) => {
+  //get addToCart then listen to click
+  document.getElementById("addToCart").addEventListener("click", (event) => {
+    event.preventDefault();
+    //get quantity value
+    product.quantity = parseInt(document.getElementById("quantity").value);
+    //get color value
+    product.color = document.getElementById("colors").value;
+    //get array of new product
+    let productArray = {
+      _id: product._id,
+      quantity: product.quantity,
+      color: product.color,
+    };
+    //Check quantity is less than 100 and higher than 0
+    if (product.quantity <= 0 || product.quantity > 100) {
+      alert("Check quantity ( not more than 100)")
```

Comment:
Use french because it's for client : "La quantité doit être supérieure à 0 et inférieure à 100)